### DISCUSSION TITLE
Js compressed and for reading

### DISCUSSION THREAD

**Discussion by Alexufo** [emscripten-core/emscripten#21668] [Q&A] [answered]
How to generate two js files: compressed and for reading with comments?

**Comment by sbc100** [comment]
You can use `--minify=0` if you want to preserve comments.

To get the maximum minification you would use `-Oz` and `--closure=1`.

**Comment by Alexufo** [comment]
@sbc100  thanks! Is it possible to do without double running emscripten? I want to distribute a solution with wasm.js and wasm.min.js.
I think this can only be done with a third party js compression tool

**Reply by sbc100** [reply, marked as answer]
No, we don't support that today.  If you want to distribute two different versions of the same program you currently need to run at least the link step twice.   Is there some reason you can't do this?  Or does your like step take a very long time maybe/

**Reply by Alexufo** [reply]
Yes, the project takes a long time to compile. I would like to leave an opportunity to study a code and to use it in production. 
I got it! Thanks!

**Reply by sbc100** [reply]
(Does you program that a long time to like too?  You wouldn't need to compile twice in this case, only link twice)
